odoo/models.py

```diff
@@ -4983,9 +4983,15 @@ def create(self, vals_list: list[ValuesType]) -> Self:
                 inv_names = {field.name for field in fields}
                 rec_vals = [
                     (data['record'], {
-                        name: data['inversed'][name]
-                        for name in inv_names
-                        if name in data['inversed']
+                        fname: data['inversed'][fname]
+                        for field in fields
+                        if (
+                            (fname := field.name) in data['inversed'] and
+                            # Don't re-update stored protected cache values because
+                            # we are sure that the cache has been already set in _create and
+                            # to avoid cache inconstency for X2many commands
+                            not (fname in data['stored'] and field in data['protected'])
+                        )
```

Comment:
This change looks like a technical hack that I can't make sense of.
Maybe we should compare this with how method `write()` deals with inverse methods.
In `write()`:
* we don't protect modified non-stored x2many fields;
* we set the field in cache before calling its inverse method only if the field is not stored and is missing in cache.
Maybe it would be better to only update the cache if it is empty, don't you think?

Reply:
Indeed, my approach is quite brittle.

> we set the field in cache before calling its inverse method only if the field is not stored and is missing in cache.
> Maybe it would be better to only update the cache if it is empty, don't you think?

It makes much more sense, I tried something like that, but `test_check_label` test was broken because of the `check_number` field that is a compute, inverse, store, readonly field (does it make any sense ?). It fails because during the creation readonly field aren't protected, then marks as recompute but still having a value in cache because it was stored. Should I changed it (or change the test which doesn't make any sense to it), or you think it is too risky for the stable ?

We can discuss about it tomorrow, it will be easier.